thread-manager/Cargo.toml

```diff
@@ -0,0 +1,38 @@
+[package]
+name = "agave-thread-manager"
+description = "Thread pool manager for agave"
+
+version = { workspace = true }
+authors = { workspace = true }
+repository = { workspace = true }
+homepage = { workspace = true }
+license = { workspace = true }
+edition = { workspace = true }
+
+publish = false
+
+[dependencies]
+anyhow = { workspace = true }
+cfg-if = { workspace = true }
+log = { workspace = true }
+num_cpus = { workspace = true }
+rayon = { workspace = true }
+serde = { workspace = true, features = ["derive"] }
+solana-metrics = { workspace = true }
+thread-priority = { workspace = true }
+tokio = { workspace = true, features = ["time", "rt-multi-thread"] }
+
+[target.'cfg(target_os = "linux")'.dependencies]
+affinity = "0.1.2"
+
+[dev-dependencies]
+agave-thread-manager = { path = ".", features = ["dev-context-only-utils"] }
+axum = "0.7.9"
+env_logger = { workspace = true }
+hyper = { workspace = true, features = ["http1", "client", "stream", "tcp"] }
+serde_json = { workspace = true }
+toml = { workspace = true }
+tower = "0.5.2"
```

Comment:
I think same here for the versions being in the top-level Cargo.

Reply:
Since these are not for tests but rather for examples only, I do bot think slowing down CI for everyone is justified. Axum is not exactly small.

Reply:
dependency trees wouldn't change though, right? i don't think it'd download/build it unless you actually tried to build one of the examples using it. it'd just prevent us from accidently having another example use axum somewhere but using a different version, and now we would need to do download/build twice

Reply:
Just checked, it seems that it does not matter if its added to workspace or not. Moved to workspace.